src/App.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import './App.css';
+import { getTodos, getUsers } from './api';
+import { ToDoList } from './components/ToDoList';
+import { Button } from 'semantic-ui-react';
+import { sortTodos } from './components/sortTodos';
+import { Users } from './components/Users';
+class  App extends React.Component {
+  state = {
+    todos: [],
+    users: [],
+    load: false,
+    buttonInnerText: 'Load Data',
+    disableButton: false
+  }
+  sortedList = (array,sortField) => {
+    this.setState({ todos: sortTodos(array, sortField) })
+  }
+  async loadInformation() {
+    this.setState({
+      disableButton: true,
+      buttonInnerText: "Loading...",
+    });
+    const [ todos, users ] = await Promise.all([
+      getTodos(),
+      getUsers(),
+    ]);
+    const fullArr = [ ...todos];
+    fullArr.map(el => el.user = users.find(user => user.id === el.userId));
+    this.setState({
+      todos: fullArr,
+      users,
+      load: true
+    });
+  }
+  handleCheck = (obj) => {
+    const copy = { ...obj };
+    const copyTodos  = [ ...this.state.todos ];
+    
+
+    let completedTask = copyTodos.find(el => JSON.stringify(el) === JSON.stringify(copy)).completed;
+     copyTodos.find(el => JSON.stringify(el) === JSON.stringify(copy)).completed = !completedTask;
```

Comment:
These methods have to be redone. First of all please read this article:
https://medium.com/ft-product-technology/this-one-line-of-javascript-made-ft-com-10-times-slower-5afb02bfd93f
Suggestion: find elements by `id`, not by comparison of stringified objects.

Reply:
done